src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_DeconstructionAssignmentOperator.cs

```diff
@@ -105,12 +105,20 @@ private void ApplyAssignments(BoundDeconstructionAssignmentOperator node, ArrayB
                 var assignmentInfo = node.AssignmentSteps[i];
                 AddPlaceholderReplacement(assignmentInfo.OutputPlaceholder, lhsTargets[i]);
 
-                // All the input placeholders for the assignments should already be set
-                var assignment = VisitExpression(assignmentInfo.Assignment);
+                var assignment = assignmentInfo.Assignment;
+
+                // All the input placeholders for the assignments should already be set with lowered nodes
+                Debug.Assert(assignment.Left.Kind == BoundKind.DeconstructValuePlaceholder);
+                Debug.Assert(assignment.Right.Kind == BoundKind.DeconstructValuePlaceholder);
+                var rewrittenLeft = (BoundExpression)Visit(assignment.Left);
+                var rewrittenRight = (BoundExpression)Visit(assignment.Right);
+
+                var loweredAssignment = MakeAssignmentOperator(assignment.Syntax, rewrittenLeft, rewrittenRight, assignment.Type,
+                                            used: true, isChecked: false, isCompoundAssignment: false);
```

Comment:
I'm not sure. I'm using the same option that the previous code was using. The local rewriter's `VisitAssignmentOperator` is implicitly setting `isChecked` to false (by omission) when it calls `MakeDynamicSetIndex` or `MakeDynamicSetMember`.